moshi/src/main/resources/META-INF/proguard/moshi.pro

```diff
@@ -0,0 +1,18 @@
+# JSR 305 annotations are for embedding nullability information.
+-dontwarn javax.annotation.**
+
+-keepclasseswithmembers class * {
+    @com.squareup.moshi.* <methods>;
+}
+
+-keep @com.squareup.moshi.JsonQualifier interface *
+
+# The name of @JsonClass types is used to look up the generated adapter
+-keepnames @com.squareup.moshi.JsonClass class *
+
+# Retain generated JsonAdapters if annotated type is retained
+-if @com.squareup.moshi.JsonClass class *
+-keep class <1>JsonAdapter {
```

Comment:
lazyweb: what does the `<1>` mean?

Reply:
It matches the first wildcard in the `if`. In this case it's the entire class name.

Reply:
Nice. Thanks.